storage/bucket/main.tf

```diff
@@ -1,35 +1,60 @@
-resource "google_storage_bucket" "bucket" {
-  name                        = "${var.project}-${var.name}"
-  project                     = var.project
-  location                    = var.location
-  force_destroy               = true
-  uniform_bucket_level_access = true
-  public_access_prevention    = "enforced"
-  labels                      = var.labels
+resource "google_storage_bucket" "buckets" {
+  for_each = { for k, v in var.buckets : k => v if v.name != "" }
 
-  versioning {
-    enabled = var.versioning
+  name               = each.value.name
+  location           = each.value.location
+  storage_class      = each.value.storage_class
+  force_destroy      = each.value.force_destroy
+  uniform_bucket_level_access = each.value.uniform_bucket_level_access
+
+  labels = each.value.labels
+
+  dynamic "lifecycle_rule" {
+    for_each = each.value.lifecycle_rule != null ? [each.value.lifecycle_rule] : []
+    content {
+      action {
+        type = lifecycle_rule.value.action.type
+        storage_class = lifecycle_rule.value.action.storage_class
+      }
+      condition {
+        age = lifecycle_rule.value.condition.age
+        created_before = lifecycle_rule.value.condition.created_before
+        with_state = lifecycle_rule.value.condition.with_state
+        matches_storage_class = lifecycle_rule.value.condition.matches_storage_class
+      }
+    }
   }
 
-  encryption {
-    default_kms_key_name = var.kms_key_id
+  dynamic "cors" {
```

Comment:
We should be mandating more in this module and not just accepting the configuration.